### PR TITLE
add snippy/core module

### PR DESCRIPTION
**Pull request by rpetit3**
## PR checklist

Closes https://github.com/nf-core/modules/issues/1854

- [x] This comment contains a description of changes (with reason).
- [x] If you've fixed a bug or added code that should be tested, add tests!
- [x] If you've added a new tool - have you followed the module conventions in the [contribution docs](https://github.com/nf-core/modules/tree/master/.github/CONTRIBUTING.md)
- [x] If necessary, include test data in your PR.
- [x] Remove all TODO statements.
- [x] Emit the `versions.yml` file.
- [x] Follow the naming conventions.
- [x] Follow the parameters requirements.
- [x] Follow the input/output options guidelines.
- [x] Add a resource `label`
- [x] Use BioConda and BioContainers if possible to fulfil software requirements.
- Ensure that the test works with either Docker / Singularity. Conda CI tests can be quite flaky:
  - [x] `PROFILE=docker pytest --tag <MODULE> --symlink --keep-workflow-wd --git-aware`
  - [x] `PROFILE=singularity pytest --tag <MODULE> --symlink --keep-workflow-wd --git-aware`
  - [x] `PROFILE=conda pytest --tag <MODULE> --symlink --keep-workflow-wd --git-aware`

### CODE REVIEW

File: modules/snippy/core/main.nf [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+    tuple val(meta), path("${prefix}.vcf")     , emit: vcf
+    tuple val(meta), path("${prefix}.txt")     , emit: txt
+    path "versions.yml"                        , emit: versions
+
+    when:
+    task.ext.when == null || task.ext.when
+
+    script:
+    def args = task.ext.args ?: ''
+    prefix = task.ext.prefix ?: "${meta.id}"
```
Comment (jfy133) [inline modules/snippy/core/main.nf:27]:
```suggestion
    def prefix = task.ext.prefix ?: "${meta.id}"
```

File: modules/snippy/core/main.nf [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+    tuple val(meta), path(vcf), path(aligned_fa)
+    path reference
+
+    output:
+    tuple val(meta), path("${prefix}.aln")     , emit: aln
+    tuple val(meta), path("${prefix}.full.aln"), emit: full_aln
+    tuple val(meta), path("${prefix}.tab")     , emit: tab
+    tuple val(meta), path("${prefix}.vcf")     , emit: vcf
+    tuple val(meta), path("${prefix}.txt")     , emit: txt
+    path "versions.yml"                        , emit: versions
```
Comment (jfy133) [inline modules/snippy/core/main.nf:20]: Ideally these would be gzipped, or does this block downstream snippy tools/steps?

File: modules/snippy/core/meta.yml [hunk 3]
```diff
@@ -0,0 +1,66 @@
+name: snippy_core
+description: Core-SNP alignment from Snippy outputs
+keywords:
+  - core
+  - alignment
+  - bacteria
```
Comment (jfy133) [inline modules/snippy/core/meta.yml:6]:
```suggestion
  - bacteria
  - snippy
```

File: tests/modules/snippy/core/main.nf [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+workflow test_snippy_core {
+    
+    input = [ [ id:'test', single_end:false ], // meta map
+              [ file(params.test_data['candidatus_portiera_aleyrodidarum']['illumina']['test_1_fastq_gz'], checkIfExists: true),
+                file(params.test_data['candidatus_portiera_aleyrodidarum']['illumina']['test_2_fastq_gz'], checkIfExists: true) ]
+            ]
+    reference = file(params.test_data['candidatus_portiera_aleyrodidarum']['genome']['genome_fasta'], checkIfExists: true)
+
+    SNIPPY_RUN ( input, reference )
+    SNIPPY_RUN.out.vcf.join(SNIPPY_RUN.out.aligned_fa).map{ meta, vcf, aln -> [[id:'snippy-core'], vcf, aln] }.set{ ch_snippy_core }
```
Comment (jfy133) [inline tests/modules/snippy/core/main.nf:17]:
```suggestion
    SNIPPY_RUN.out.vcf.join( SNIPPY_RUN.out.aligned_fa ).map{ meta, vcf, aln -> [[id:'snippy-core'], vcf, aln] }.set{ ch_snippy_core }
```

File: tests/modules/snippy/core/main.nf [hunk 5]
```diff
@@ -0,0 +1,20 @@
...
+    SNIPPY_CORE(ch_snippy_core, reference)
```
Comment (jfy133) [inline tests/modules/snippy/core/main.nf:18]:
```suggestion
    SNIPPY_CORE( ch_snippy_core, reference )
```

File: modules/snippy/core/main.nf [hunk 6]
```diff
@@ -0,0 +1,53 @@
...
+    # Run snippy-core
+    snippy-core \\
+        $args \\
+        --ref $reference_name \\
+        --prefix $prefix \\
+        samples/*
+
+    cat <<-END_VERSIONS > versions.yml
+    "${task.process}":
+        snippy: \$(echo \$(snippy --version 2>&1) | sed 's/snippy //')
```
Comment (jfy133) [inline modules/snippy/core/main.nf:50]:
```suggestion
        snippy: \$(echo \$(snippy-core --version 2>&1) | sed 's/snippy //')
```

**Review by jfy133 — APPROVED** [review]
LGTM, just a couple minor comments

File: modules/snippy/core/main.nf (hunk 1 above)
Comment (rpetit3) [reply modules/snippy/core/main.nf:27]: will this be ok if I'm using `$prefix` in the output block?

Comment (jfy133) [reply modules/snippy/core/main.nf:27]: Correct me if I'm wrong, but the output prefix won't be necessary as you shouldn't have any 'input' fasta/aln/tab etc. in your main directory, correct? You could just give `*fasta` `*aln` directly? Or does the output also go in `samples`?

File: modules/snippy/core/main.nf (hunk 2 above)
Comment (jfy133) [reply modules/snippy/core/main.nf:20]: Ah, well I would argue (given this is specific to a 'pipeline' of tools, so I'm guessing you're unlikely to use the output of snippy-core outside of `snippy`) you don't need to compress if the next step doesn't accept it.

File: modules/snippy/core/main.nf (hunk 1 above)
Comment (rpetit3) [reply modules/snippy/core/main.nf:27]: Think we'll have an issue with `*.aln` capturing both `${prefix}.aln` and `${prefix}.full.aln` ?

File: modules/snippy/core/main.nf (hunk 2 above)
Comment (rpetit3) [reply modules/snippy/core/main.nf:20]: Sound good, thank you

File: modules/snippy/core/main.nf (hunk 1 above)
Comment (jfy133) [reply modules/snippy/core/main.nf:27]: Ah fair point. Fine as is then!

**Review by jfy133 — APPROVED** [review]

**Pull request merged**